modules/sharedaddy/sharing-service.php

```diff
@@ -189,8 +189,10 @@ public function get_blog_services() {
 		 * @see https://github.com/Automattic/jetpack/issues/6121
 		 */
 		if ( ! is_array( $options ) || ! isset( $options['button_style'], $options['global'] ) ) {
-			$global_options = $this->get_global_options();
-			$options = array_merge( is_array( $options ) ? $options : array(), $global_options );
+			$options = array_merge(
+				is_array( $options ) ? $options : array(),
+				array( 'global' => $this->get_global_options() )
+			);
```

Comment:
We shouldn't do an `array_merge` unnecessarily (with an empty array) since it's expensive. Can we do this?
```php
$options = is_array( $options )
    ? array_merge( $options, array( 'global' => $this->get_global_options() ) )
    : array( 'global' => $this->get_global_options() );
```